chainer/function_set.py

```diff
@@ -126,9 +126,12 @@ def parameters(self):
 
     @parameters.setter
     def parameters(self, params):
-        param_iter = iter(params)
+        idx = 0
         for _, func in self._get_sorted_funcs():
-            func.parameters = param_iter
+            param_size = len(func.parameters)
+            func.parameters = params[idx: idx + param_size]
+            idx += param_size
+        assert idx == len(params)
```

Comment:
Why don't you check sum of length of params before assigning? Such code is exception-safe.